api/repositories/service_binding_repository_test.go

```diff
@@ -237,7 +243,103 @@ var _ = Describe("ServiceBindingRepo", func() {
 		When("the user doesn't have permission to list ServiceBindings in the Space", func() {
 			It("returns a Forbidden error", func() {
 				_, err := repo.ServiceBindingExists(testCtx, authInfo, space.Name, appGUID, serviceInstanceGUID)
-				Expect(err).To(BeAssignableToTypeOf(ForbiddenError{}))
+				Expect(err).To(BeAssignableToTypeOf(repositories.ForbiddenError{}))
+			})
+		})
+	})
+
+	Describe("ListServiceBindings", func() {
+		var (
+			serviceBinding1, serviceBinding2 *servicesv1alpha1.CFServiceBinding
+			space2                           *hnsv1alpha2.SubnamespaceAnchor
+
+			serviceInstance2GUID string
+
+			requestMessage repositories.ListServiceBindingsMessage
+		)
+
+		BeforeEach(func() {
+			cfApp1 := createAppCR(testCtx, k8sClient, "app-1-name", generateGUID(), space.Name, "STOPPED")
+			cfServiceInstance1 := createServiceInstanceCR(testCtx, k8sClient, generateGUID(), space.Name, "service-instance-1-name", "secret-1-name")
+			serviceBinding1Name := "service-binding-1-name"
+			serviceBinding1 = createServiceBindingCR(testCtx, k8sClient, generateGUID(), space.Name, &serviceBinding1Name, cfServiceInstance1.Name, cfApp1.Name)
+
+			space2 = createSpaceAnchorAndNamespace(testCtx, org.Name, prefixedGUID("space2"))
+			cfApp2 := createAppCR(testCtx, k8sClient, "app-2-name", generateGUID(), space2.Name, "STOPPED")
+			serviceInstance2GUID = generateGUID()
+			cfServiceInstance2 := createServiceInstanceCR(testCtx, k8sClient, serviceInstance2GUID, space2.Name, "service-instance-2-name", "secret-2-name")
+			serviceBinding2 = createServiceBindingCR(testCtx, k8sClient, generateGUID(), space2.Name, nil, cfServiceInstance2.Name, cfApp2.Name)
+
+			requestMessage = repositories.ListServiceBindingsMessage{}
+		})
+
+		When("the user has access to both namespaces", func() {
+			BeforeEach(func() {
+				createRoleBinding(testCtx, userName, spaceDeveloperClusterRole.Name, space.Name)
+				createRoleBinding(testCtx, userName, spaceDeveloperClusterRole.Name, space2.Name)
+			})
+
+			Describe("no query parameters are specified", func() {
+				It("returns a list of ServiceBindingRecords in the spaces the user has access to", func() {
+					Eventually(func() []repositories.ServiceBindingRecord {
```

Comment:
Does this need to be `Eventually` instead of ordinary `Expect`? Resources are created synchronously in the `BeforeEach`, so should be immediately available in the `It`

Reply:
I don't know that the resources are actually synchronously available, even after the create call succeeds,  though perhaps the cache gets updated if the element is absent? I feel like we've seen flakes related to directly calling `Expect`.